### PR TITLE
12792 migrate tests from  web webhook test.go to use testify

### LINKED ISSUE #12792: Migrate tests from "web/webhook_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `web/webhook_test.go` file.

The expected way to migrate this is to go to the `web/webhook_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19463

### PR DESCRIPTION
**Pull request by sascha-andres**
#### Summary

Migrate tests from "web/webhook_test.go" to use testify

#### Ticket Link

[MM-19463](https://mattermost.atlassian.net/browse/MM-19463)

Closes #12792

### CODE REVIEW

File: web/webhook_test.go [hunk 1]
```diff
@@ -263,12 +261,11 @@ func TestCommandWebhooks(t *testing.T) {
 	assert.Equal(t, http.StatusBadRequest, resp.StatusCode)
 
 	for i := 0; i < 5; i++ {
-		if resp, appErr := http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`)); err != nil || resp.StatusCode != http.StatusOK {
-			t.Fatal(appErr)
-		}
+		resp, appErr := http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`))
```
Comment (hanzei) [inline web/webhook_test.go]: You are shadowing `resp` from L255 here. You can just use:
```suggestion
		resp, err = http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`))
```

Comment (sascha-andres) [reply web/webhook_test.go]: changed accordingly

**Comment by sascha-andres** [comment]
rebased on top of master

File: web/web_test.go [hunk 2]
```diff
@@ -254,9 +251,8 @@ func TestCheckClientCompatability(t *testing.T) {
 	}
 	for _, browser := range uaTestParameters {
 		t.Run(browser.Name, func(t *testing.T) {
-			if result := CheckClientCompatability(browser.UserAgent); result != browser.Result {
-				t.Fatalf("%s User Agent Test failed!", browser.Name)
-			}
+		  result := CheckClientCompatability(browser.UserAgent)
+			require.Equalf(t, result, browser.Result, "user agent test failed for %s", browser.Name)
```
Comment (mgdelacroix) [inline web/web_test.go]: nit: these two lines are not correctly aligned

File: web/webhook_test.go [hunk 3]
```diff
@@ -263,12 +261,13 @@ func TestCommandWebhooks(t *testing.T) {
 	assert.Equal(t, http.StatusBadRequest, resp.StatusCode)
 
 	for i := 0; i < 5; i++ {
-		if resp, appErr := http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`)); err != nil || resp.StatusCode != http.StatusOK {
-			t.Fatal(appErr)
-		}
+		t.Run(fmt.Sprintf("hook #%d", i), func(t *testing.T) {
+			resp, err = http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`))
+			assert.NoErrorf(t, err, "hook #%d failed", i)
```
Comment (mgdelacroix) [inline web/webhook_test.go]: This line should use `require`, as if there is an error and if continues, the `resp` object can be `nil`:

```suggestion
			require.NoErrorf(t, err, "hook #%d failed", i)
```

File: web/web_test.go [hunk 4]
```diff
@@ -220,11 +220,8 @@ func TestStatic(t *testing.T) {
 
 	resp, err := http.Get(URL + "/static/root.html")
 
-	if err != nil {
-		t.Fatalf("got error while trying to get static files %v", err)
-	} else if resp.StatusCode != http.StatusOK {
-		t.Fatalf("couldn't get static files %v", resp.StatusCode)
-	}
+	assert.NoErrorf(t, err, "got error while trying to get static files %v", err)
```
Comment (mgdelacroix) [inline web/web_test.go]: This line should use `require` as `resp` would be `nil` if an error is returned:

```suggestion
	require.NoErrorf(t, err, "got error while trying to get static files %v", err)
```

File: web/webhook_test.go [hunk 5]
```diff
@@ -250,9 +250,7 @@ func TestCommandWebhooks(t *testing.T) {
 	}
 
 	hook, appErr := th.App.CreateCommandWebhook(cmd.Id, args)
-	if appErr != nil {
-		t.Fatal(appErr)
-	}
+	assert.Error(t, appErr, "error creating command web hook")
```
Comment (hanzei) [inline web/webhook_test.go]: Please use `assert.Nil` here. `assert.Error` doesn't work well with a custom error.

File: web/webhook_test.go [hunk 6]
```diff
@@ -263,12 +261,13 @@ func TestCommandWebhooks(t *testing.T) {
...
+			resp, err = http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`))
+			require.NoErrorf(t, err, "hook #%d failed", i)
+			assert.Equalf(t, http.StatusOK, resp.StatusCode, "hook #%d failed", i)
+		})
 	}
 
-	if resp, _ := http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`)); resp.StatusCode != http.StatusBadRequest {
-		t.Fatal("expected error for sixth usage")
-	}
+	resp, _ = http.Post(ApiClient.Url+"/hooks/commands/"+hook.Id, "application/json", bytes.NewBufferString(`{"text":"this is a test"}`))
```
Comment (hanzei) [inline web/webhook_test.go]: Please check the returned error

File: web/webhook_test.go [hunk 7]
```diff
@@ -263,12 +261,13 @@ func TestCommandWebhooks(t *testing.T) {
...
+	assert.Equalf(t, http.StatusBadRequest, resp.StatusCode, "expected error for sixth usage")
```
Comment (hanzei) [inline web/webhook_test.go]: Nit:
```suggestion
	assert.Equal(t, http.StatusBadRequest, resp.StatusCode, "expected error for sixth usage")
```

**Comment by hanzei** [comment]
@sascha-andres Do you have questions about the feedback?

**Comment by hanzei** [comment]
Hey @sascha-andres,

Please note that we will close this PR after another 10 days of inactivity. If you don't object, other contributors might complete this PR for you. Let me know, if you have any questions.

**Comment by jasonblais** [comment]
Given there hasn't been recent activity on this PR, we'll close this following our [inactive contributions process](https://developers.mattermost.com/contribute/getting-started/inactive-contributions/).

If you'd like to continue working on it, let us know! We'd be happy to guide you through the feedback above.

**Pull request closed without merging**